contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.c

```diff
@@ -54,35 +54,38 @@ void initialize_next_msg(int fd) {
 }
 
 int iprobe_next_msg(struct p2p_log_msg *p2p_msg) {
-  /* FIXME:  This isn't comiling yet.
+  int rc = 0;
   if (!next_msg) {
-    initialize_next_msg(fd);
+    rc = get_next_msg(NULL);
+    if (rc == 1) return -1;
   }
-  */
   if (p2p_msg) {
     *p2p_msg = next_msg_entry;
   }
   return (next_msg_entry.comm != MPI_COMM_NULL);
 }
 
-void get_next_msg(struct p2p_log_msg *p2p_msg) {
-  static int fd = -2;
-  if (fd == -2) {
+int get_next_msg(struct p2p_log_msg *p2p_msg) {
+  static int fd = -1;
+  int rc = 1;
+  if (fd == -1) {
```

Comment:
Please revert to using `-2` (or `UNINITIALIZED`) here.  If I were more careful, I would have written:
```
#define UNINITIALIZED -2
static int fd = UNINITIALIZED;
...
if (fd == UNINITALIZED) {
```

There is a big difference between `UNINITIALIZED` and `-1` (meaning that some syscall had failed.